packages/block-editor/src/components/block-list/block.js

```diff
@@ -50,6 +50,32 @@ import useMovingAnimation from './moving-animation';
 import { ChildToolbar, ChildToolbarSlot } from './block-child-toolbar';
 import { Context } from './root-container';
 
+/**
+ * A debounced version of getAccessibleBlockLabel, avoids unnecessary updates to the aria-label attribute
+ * when typing in some blocks, like the paragraph.
+ *
+ * @param {Object} blockType      The block type object representing the block's definition.
+ * @param {Object} attributes     The block's attribute values.
+ * @param {number} index          The index of the block in the block list.
+ * @param {string} moverDirection A string representing whether the movers are displayed vertically or horizontally.
+ * @param {number} delay          The debounce delay.
+ */
+const useDebouncedAccessibleBlockLabel = ( blockType, attributes, index, moverDirection, delay ) => {
+	const [ blockLabel, setBlockLabel ] = useState( '' );
+
+	useEffect( () => {
+		const timeoutId = setTimeout( () => {
+			setBlockLabel( getAccessibleBlockLabel( blockType, attributes, index + 1, moverDirection ) );
+		}, delay );
+
+		return () => {
+			clearTimeout( timeoutId );
+		};
+	}, [ blockType, attributes, index, moverDirection, delay ] );
+
+	return blockLabel;
+};
+
 function BlockListBlock( {
```

Comment:
This is one of the most central components of our codebase and changes here can have big performance impacts, so, a sanity check (running performance tests) would be good (with and without the PR.